test/functional/rpc_fundrawtransaction.py

```diff
@@ -181,6 +182,41 @@ def test_simple(self):
         dec_tx  = self.nodes[2].decoderawtransaction(rawtxfund['hex'])
         assert len(dec_tx['vin']) > 0  #test that we have enough inputs
 
+    def check_segwit_inputs(self, vins):
+        for vin in vins:
+            # check vin is a segwit input
+            utxo = self.nodes[2].gettxout(vin['txid'], vin['vout'])
+            info = self.nodes[2].getaddressinfo(utxo['scriptPubKey']['address'])
+            if not info['iswitness']:
+                return False
+
+        return True
+
+    def test_witness_only(self):
```

Comment:
it would be great to test the case where we filter inputs to a particular type (or multiple types) and then verify that the inputs in the funded tx only contain those types. you can look at `test/functioanal/wallet_avoid_mixing_output_types.py` for an example of checking the inputs to a transaction and verifying they are all of a certain `OutputType`

Reply:
Thanks, I've added these tests.
I included some helpers from `wallet_avoid_mixing_output_types` to the test framework to reuse them.